service/service.go

```diff
@@ -55,10 +55,24 @@ type Owner interface {
 
 	AddModules(modules ...ModuleCreateFunc) error
 
-	Start(waitForExit bool) (exit <-chan Exit, ready <-chan struct{}, err error)
+	// Start service is used for blocking the call on service start. Start will block the
+	// call and yield the control to the service lifecyce manager.
+	Start() Control
```

Comment:
if the start is blocking, why do we return control, if it's going to block?
unless i'm misunderstanding something